### PR TITLE
feat(PVO11Y-5348): env split drift detection

### PR DESCRIPTION
**Pull request by TominoFTW**
### Description

Jira: [PVO11Y-5348](https://redhat.atlassian.net/browse/PVO11Y-5348)

The new structure:
```
rhobs/
  staging/
    alerting/
      data-plane/
        alerts/                    <-- app-interface points here @ main
          prometheus.pod_alerts.yaml
          prometheus.pipeline_alerts.yaml
          prometheus.cluster_monitoring_absent_alerts.yaml
          ...
        tests/
          pod_alerts_test.yaml
          pipeline_alerts_test.yaml
          cluster_monitoring_absent_test.yaml
          ...
      konflux-release-data/
        alerts/                    <-- app-interface points here @ main
          prometheus.krd_runner_alerts.yaml
          prometheus.krd_pipeline_alerts.yaml
          ...
        tests/
          krd_runner_alerts_test.yaml
          krd_pipeline_alerts_test.yaml
          ...
    recording/
      rules/                       <-- app-interface points here @ main
        api_server_availability_recording_rules.yaml
        exporter_recording_rules.yml
        ...
      tests/
        recording_test.yaml
        ...
  production/
    (identical structure)
```

#### Changes

**1. New `rhobs/staging/` directory** (165 files)

Copies of all existing rules and tests, created using a [history-preserving git technique](https://stackoverflow.com/a/44036771) so \`git log --follow\` and \`git blame\` trace back to original authors and commits.

| Source | Destination |
|--------|-------------|
| \`rhobs/alerting/data_plane/\` | \`rhobs/staging/alerting/data-plane/alerts/\` |
| \`rhobs/alerting/konflux-release-data/\` | \`rhobs/staging/alerting/konflux-release-data/alerts/\` |
| \`rhobs/recording/\` | \`rhobs/staging/recording/rules/\` |
| \`test/promql/tests/data_plane/\` | \`rhobs/staging/alerting/data-plane/tests/\` |
| \`test/promql/tests/konflux-release-data/\` | \`rhobs/staging/alerting/konflux-release-data/tests/\` |
| \`test/promql/tests/recording/\` | \`rhobs/staging/recording/tests/\` |

Original directories are **untouched** - they still exist until production is added and app-interface ready

**2. Drift detection tool** (\`scripts/check-env-drift.py\`, 954 lines)

Compares \`rhobs/staging/\` vs \`rhobs/production/\` with direction-aware severity:
- **Staging-only** file/rule -> warning (staged rollout is normal)
- **Production-only** file/rule -> violation (production must not drift ahead)
- **Field-level diff** -> violation (unless bypassed)

Bypass mechanism via inline comments:
- \`# drift:ignore <reason>\` — rule-level or field-level bypass
- \`# drift:ignore-file <reason>\` — entire file bypass

**3. GitHub Actions workflows**

| Workflow | Trigger | Purpose |
|----------|---------|---------|
| \`check-drift.yaml\` | \`pull_request\` (paths: \`rhobs/**\`) | Runs drift detection, uploads artifact |
| \`check-drift-comment.yaml\` | \`workflow_run\` | Posts/updates PR comment with results (fork-safe) |
| \`weekly-drift-report.yaml\` | Weekly cron + manual | Full scan with drift age analysis |

The two-workflow pattern for PR comments uses \`workflow_run\` to keep write permissions separated from PR code execution (fork safety).

**4. Test suite** (29 scenarios)

Covers file-level, rule-level, field-level diffs, bypass mechanisms, edge cases (\`drift:ignore\` placement, empty files, duplicate rule names across groups), flag interactions (\`--strict\`, \`--allow-fail\`), recording rules, and report mode.

**5. Documentation & integration**

- \`docs/ALERT_DRIFT.md\` - documentation of the drift detection system
- Updated \`README.md\`, \`CONTRIBUTING.md\`, \`AGENTS.md\` - path references
- Updated \`Makefile\` - \`check-env-drift\` target added to \`make all\`
- Updated \`scripts/check-alert-conventions.py\` - scans staging/production alerting dirs

#### Rollout plan

1. **This PR**: staging directory + drift detection tooling
2. **Follow-up PR**: production directory (same git technique, drift detection becomes active)
3. **app-interface update**: point refs at new paths
4. **Cleanup PR**: remove old directories

[PVO11Y-5348]: https://redhat.atlassian.net/browse/PVO11Y-5348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/check-drift-comment.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Post PR comment
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          report=$(sed '/^$/d' drift-report.txt | sed 's/```/` ` `/g')
+          pr_number=$(cat pr_number)
+          exit_code=$(cat exit_code)
```
[the remark this answers is not in the document]
Comment (TominoFTW) [reply .github/workflows/check-drift-comment.yaml:31]: Not really concern as nothing would be happening, worst case misleading comment...

**Review by raks-tt — CHANGES_REQUESTED** [review]
/hold

**Comment by TominoFTW** [comment]
Some examples of messages in the PRs:
Failed + blocks pipeline:
### Drift Detection ❌ Failed

Comparing `d7cfc9c` → `e24b630`
```
--- Drift violations (1) ---
  [DRIFT] production/alerting/data-plane/alerts/prometheus.pipeline_alerts.yaml: rule 'pipeline_alerts/PipelineRunStuckPending' — exists only in production
```

Warning + pipeline passes:
### Drift Detection ⚠️ Warning

Comparing `d7cfc9c` → `58f1c71`
```
--- Warnings (1) ---
  [WARN] staging/alerting/data-plane/alerts/prometheus.pipeline_alerts.yaml: rule 'pipeline_alerts/PipelineRunStuckPending' — exists only in staging
--- No drift violations found ---
check-env-drift: OK
```

Acknowledged bypass (# drift:ignore comment):
### Drift Detection ✅ Passed

Comparing `d7cfc9c` → `3bd8106`
```
--- Acknowledged bypasses (1) ---
  [bypass] alerting/data-plane/alerts/prometheus.pipeline_alerts.yaml: rule 'pipeline_alerts/PipelineRunStuckPending' — staging-first rollout, will promote to production after validation
--- No drift violations found ---
check-env-drift: OK
```

File: scripts/check-alert-conventions.py [hunk 2]
```diff
@@ -29,9 +32,11 @@ def check_rules(path):
     return errors
 
 def main():
-    files = sorted(glob.glob(f"{ALERT_DIR}/**/*.yaml", recursive=True))
+    files = []
```
[the remark this answers is not in the document]
Comment (TominoFTW) [reply scripts/check-alert-conventions.py:35]: not harming

File: Makefile [hunk 3]
```diff
@@ -24,6 +24,17 @@ check-and-test:
 	$(CMD) -t rhtap -d rhobs/alerting/data_plane -y -p --tests-dir test/promql/tests/data_plane
 	$(CMD) -t rhtap -d rhobs/alerting/konflux-release-data -y -p --tests-dir test/promql/tests/konflux-release-data
 	$(CMD) -t rhtap -d rhobs/recording -y -p --tests-dir test/promql/tests/recording
+	$(CMD) -t rhtap -d rhobs/staging/alerting/data-plane/alerts -y -p --tests-dir rhobs/staging/alerting/data-plane/tests
+	$(CMD) -t rhtap -d rhobs/staging/alerting/konflux-release-data/alerts -y -p --tests-dir rhobs/staging/alerting/konflux-release-data/tests
+	$(CMD) -t rhtap -d rhobs/staging/recording/rules -y -p --tests-dir rhobs/staging/recording/tests
+
+.PHONY: check-env-drift
+check-env-drift:
```
[the remark this answers is not in the document]
Comment (TominoFTW) [reply Makefile:32]: Technically correct, but it is being installed as it is dependency of yamllint.

**Review by raks-tt — APPROVED** [review]
/lgtm

**Review by raks-tt — APPROVED** [review]
